system_metrics_collector/src/system_metrics_collector/collector.hpp

```diff
@@ -29,7 +30,7 @@ namespace system_metrics_collector
 /**
  * Simple class in order to collect observed data and generate statistics for the given observations.
  */
-class Collector
+class Collector : virtual public MetricDetailsInterface
```

Comment:
I think this is a good change, but leaves us with some cleanup. See other comments.

Reply:
See comment about virtual inheritance.